lib/markdown.dart

```diff
@@ -2,7 +2,36 @@
 // for details. All rights reserved. Use of this source code is governed by a
 // BSD-style license that can be found in the LICENSE file.
 
-/// Parses text in a Markdown-like format and renders to HTML.
+/// Parses text in a Markdown-like format building an
+/// [AST tree](https://en.wikipedia.org/wiki/Abstract_syntax_tree)
+/// that can then be rendered to HTML.
+///
+/// If you are only interested in rendering Markdown to HTML please refer
+/// to the [README](../index.html) which explains the use of [markdownToHtml()].
+///
+/// The main entrypoint to the library is the [Document] which encapsulates the
+/// parsing process converting a Markdown text into a tree of [Node] (`List<Node>`).
+///
+/// Two main parsing mechanics are used:
+///
+/// - Blocks, representing top level elements like: headers, paragraphs, blockquotes,
+///   code blocks, ... implemented via [BlockSyntax] subclasses.
+/// - Inlines, representing chunks of text within a block with special meaning, like:
+///   links, emphasis, inlined code, ... implemented via [InlineSyntax] subclasses.
+///
+/// Looking closely at [Document.Document()] a few other concepts merit a mention:
```

Comment:
This is a deprecated way to refer to an unnamed constructor. Could you change to `[Document.new]` perhaps?

Reply:
done